drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1184 @@
+
+	if (retval < 0) {
+		LOG_DBG("error setting SENSOR_CONFIG_1");
+		return -EIO;
+	}
+
+	/* REG_SENSOR_CONFIG_2 */
+	regdata = 0;
+
+	val = (drv_cfg->ch_mag_gain_correction == TMAG5273_CORRECTION_CH_2)
+		? TMAG5273_MAG_GAIN_CORRECTION_CH_2 : TMAG5273_MAG_GAIN_CORRECTION_CH_1;
+
+	regdata = TMAG5273_FIELD_SET(regdata, TMAG5273_GAIN_CORRECTION_CH_MSK, val);
+
+	switch (drv_cfg->angle_magnitude_axis) {
+	case TMAG5273_ANGLE_MAG_XY: {
+		val = TMAG5273_ANGLE_EN_XY;
+		break;
+	}
+	case TMAG5273_ANGLE_MAG_YZ: {
+		val = TMAG5273_ANGLE_EN_YZ;
+		break;
+	}
+	case TMAG5273_ANGLE_MAG_XZ: {
+		val = TMAG5273_ANGLE_EN_XZ;
+		break;
+	}
+	case TMAG5273_ANGLE_MAG_RUNTIME: {
+		__fallthrough;
+	}
+	case TMAG5273_ANGLE_MAG_NONE: {
+		__fallthrough;
+	}
+	default: {
+		val = TMAG5273_ANGLE_EN_NONE;
+		break;
+	}
+	}
+
+	regdata = TMAG5273_FIELD_SET(regdata, TMAG5273_MAG_CH_EN_MSK, val);
+
+	/* measurment range "high" by default */
+	switch (drv_cfg->meas_range) {
+	case TMAG5273_AXIS_RANGE_LOW: {
+		regdata = TMAG5273_FIELD_SET(regdata, TMAG5273_X_Y_RANGE_MSK,
+			TMAG5273_XYZ_MEAS_RANGE_LOW);
+		regdata = TMAG5273_FIELD_SET(regdata, TMAG5273_Z_RANGE_MSK,
+			TMAG5273_XYZ_MEAS_RANGE_LOW);
+		break;
+	}
+	case TMAG5273_AXIS_RANGE_HIGH: {
+		__fallthrough;
+	}
+	case TMAG5273_AXIS_RANGE_RUNTIME: {
+		__fallthrough;
+	}
+	default: {
+		regdata = TMAG5273_FIELD_SET(regdata, TMAG5273_X_Y_RANGE_MSK,
+			TMAG5273_XYZ_MEAS_RANGE_HIGH);
+		regdata = TMAG5273_FIELD_SET(regdata, TMAG5273_Z_RANGE_MSK,
+			TMAG5273_XYZ_MEAS_RANGE_HIGH);
+		break;
+	}
+	}
+
+	retval = i2c_reg_write_byte_dt(&drv_cfg->i2c, TMAG5273_REG_SENSOR_CONFIG_2, regdata);
+
+	if (retval < 0) {
+		LOG_DBG("error setting SENSOR_CONFIG_2");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief initialize a TMAG5273 sensor
+ *
+ * @param dev handle to the device
+ *
+ * @retval \c 0 on success
+ * @retval \c -EINVAL if bus label is invalid
+ * @retval \c -EIO on communication errors
+ */
+static int tmag5273_init(const struct device *dev)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	struct tmag5273_data *drv_data = dev->data;
+
+	if (!i2c_is_ready_dt(&drv_cfg->i2c)) {
+		LOG_DBG("Could not get pointer to TMAG5273 I2C device");
+		return -EINVAL;
+	}
+
+	int retval = 0;
+
+	if (drv_cfg->need_int_pin) {
+		/* init interrupt GPIO if present */
+		if (!gpio_is_ready_dt(&drv_cfg->int_gpio)) {
+			LOG_DBG("missing gpio interrupt");
+			return -ENOTSUP;
+		}
+
+		retval = _tmag5273_int_gpio_config(dev, GPIO_INPUT);
+
+		if (retval < 0) {
+			LOG_DBG("Cannot configure GPIO |%d|", retval);
+			return -EINVAL;
+		}
+	}
+
+	uint8_t regdata = 0;
+
+	retval = i2c_reg_read_byte_dt(&drv_cfg->i2c,
+		TMAG5273_REG_DEVICE_CONFIG_2,
+		&regdata);
+
+	if (retval < 0) {
+		LOG_DBG("Could not read device config 2 register |%d|", retval);
+		return -EIO;
+	}
+
+	LOG_DBG("operation mode: %d", FIELD_GET(regdata, TMAG5273_OPERATING_MODE_MSK));
```

Comment:
I get a `Division by zero` here, because you switched the fields for mask and data. (Is the same for all FIELD_GET calls)
```
#define FIELD_GET(mask, value)  (((value) & (mask)) / LSB_GET(mask))
```

```suggestion
	LOG_DBG("operation mode: %d", FIELD_GET(TMAG5273_OPERATING_MODE_MSK, regdata));
```

Reply:
Well, this is then wrong on all occurences and needs to be fixed globally.